### ISSUE TITLE
Unable to compute the gradient of the wav2vec encoder

### ISSUE THREAD

**Issue by turian** [pytorch/fairseq#2789]
### Before asking:   

I found one closed issue on the gradient, but with no follow-up.

#### What is your question?

I cannot backprop the gradient through the wav2vec encoder. I have isolated that using the wav2vec encoder no longer backprops the gradient.

#### Code

Here is the code I run:

```
import os
import torch
import torch.tensor as T
from fairseq.models.wav2vec import Wav2VecModel

MODEL = "wav2vec_small.pt"
fname = MODEL
os.system("wget -c https://dl.fbaipublicfiles.com/fairseq/wav2vec/%s" % MODEL)
cp = torch.load(fname)
model = Wav2VecModel.build_model(cp['args'], None)
model.load_state_dict(cp['model'], strict=False)
model.eval()

time = torch.linspace(0.0, 2.5, 64000)
def sine(f0):
  return T(0.5) * torch.cos(f0 * 2.0 * 3.14 * time)

def extractor(x):
  return model.feature_extractor(torch.tensor(x.view(1, -1)))
def wav2vecdist(x, y):
  return torch.mean(torch.abs(extractor(x) - extractor(y)))

ft = T(440.0)
# If you enable GPU, you have to add 'device="cuda"'
with torch.enable_grad():
  fp = T(660.0, requires_grad=True)
  fp.retain_grad()
  dist = torch.mean(torch.abs(extractor(sine(ft)) - extractor(sine(fp))))
  print(dist)
  dist.backward(retain_graph=False)
  print(fp.grad)
```

As you can see, the extractor in wav2vec is not passing a gradient and I'm not sure why.

A minimal google colab is here: https://colab.research.google.com/drive/1l61syPS7GO7O-S0o8cWnv8R0JwzVq58T?usp=sharing

#### What have you tried?

I have tried to remove as much exterior code as possible.

I have tried fiddling with .train() and .eval() on the wav2vec model.

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.6.0+cu101
 - OS (e.g., Linux): Ubuntu 18.04.5 LTS
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): !pip install -q --editable ./
 - Python version: 3.6.9
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: Tesla P100
 - Any other relevant information: Google Colab

**Comment by turian** [comment]
I have tried also with fairseq 0.10.1 and have the same issue.

I tried further with Wav2Vec2Model and have the same issue.

**Comment by turian** [comment]
```
  return model.feature_extractor(torch.tensor(x.view(1, -1)))
```

creates a copy, which loses the gradient. The line should be:

```
  return model.feature_extractor(x.view(1, -1))
```

The colab is updated.
